modules/repository/license.go

```diff
@@ -6,13 +6,89 @@ package repository
 import (
 	"bufio"
 	"bytes"
+	"context"
 	"fmt"
 	"regexp"
 	"strings"
 
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/modules/container"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/options"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/util"
+
+	licenseclassifier "github.com/google/licenseclassifier/v2"
+)
+
+var (
+	classifier   *licenseclassifier.Classifier
+	sameLicenses map[string]string
 )
 
+func init() {
+	err := loadSameLicenses()
+	if err != nil {
+		log.Error("loadSameLicenses: %v", err)
+	}
+	err = initClassifier()
+	if err != nil {
+		log.Error("initClassifier: %v", err)
+	}
```

Comment:
I would suggest do not do it in `init()`

Gitea binary has various sub-commands. For example, we do not want to waste CPU/Memory when the Gitea binary is used as a git hook.

Reply:
Done in https://github.com/go-gitea/gitea/pull/24872/commits/c57d5422454eb46f5558892a2c20f8ac1f14c872.